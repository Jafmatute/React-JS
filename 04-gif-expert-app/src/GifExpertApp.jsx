const GifExpertApp = (props) => (
    <>
    <h1>GifExpertApp </h1>
    </>
);

export default GifExpertApp;