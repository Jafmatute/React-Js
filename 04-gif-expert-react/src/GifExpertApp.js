/** @format */

import { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({ defaultCategorias = [] }) => {
  // const categoriasList = ["One Punch"];

  const [categorias, setCategorias] = useState(defaultCategorias);

  const handleAdd = () => {
    // setCategorias([...categorias, "one push"]);
    setCategorias((cats) => [...cats, 'one puch']);
  };

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategorias={setCategorias} />
      <hr />

      <ol>
        {categorias.map((categoria) => (
          <GifGrid key={categoria} categoria={categoria} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
