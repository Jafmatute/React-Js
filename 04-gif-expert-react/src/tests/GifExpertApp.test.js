import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en el componente <GifExpertApp />', () => {
  test('Probar que el componente se renderice correctamente', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar una lista de categorÃ­as', () => {
    const categorias = ['one punch', 'Dragon ball'];
    const wrapper = shallow(<GifExpertApp defaultCategorias={categorias} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categorias.length);
  });
});
