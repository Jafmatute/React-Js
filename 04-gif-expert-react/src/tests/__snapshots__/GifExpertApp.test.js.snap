// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> Probar que el componente se renderice correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp /> debe de mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="one punch"
      key="one punch"
    />
    <GifGrid
      categoria="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;
